### PR TITLE
[GPU] Added RoPE support for ChatGLM and Qwen

### PR DESCRIPTION
**Pull request by Lyamin-Roman**
### Details:
- Added support RoPE for ChatGLM and Qwen models
- Moved and refactored RoPE functional tests

### Tickets:
 - *[119150](https://jira.devtools.intel.com/browse/CVS-119150)*

### CODE REVIEW

**Review by vladimir-paramuzov — APPROVED** [review]
LGTM except func tests

File: src/common/transformations/src/transformations/common_optimizations/fuse_rotary_positional_embeddings.cpp [hunk 1]
```diff
@@ -652,8 +653,8 @@ ov::pass::RoPEFusionQwen::RoPEFusionQwen(int split_output_id) {
     auto Multiply_567524 = makePattern<opset1::Multiply>({ShapeOf_485735, {-1}}, {{"auto_broadcast", "numpy"}});
     auto Gather_377635 = makePattern<opset8::Gather>({Multiply_567524, {1}, 0}, {{"batch_dims", 0}});
 
-    auto input_ids = makePattern("i32[?,?]");  // [batch, length]
-    auto ShapeOf_409241 = makePattern<opset1::ShapeOf>({input_ids}, {});
+    auto input_ids = makePattern();  // [batch, length]
```
Comment (vladimir-paramuzov) [inline src/common/transformations/src/transformations/common_optimizations/fuse_rotary_positional_embeddings.cpp:656]: Why did you remove `"i32[?,?]"` ?

**Comment by vladimir-paramuzov** [comment]
@Lyamin-Roman  https://github.com/openvinotoolkit/openvino/pull/24750 may cause some conflicts. Please rebase once it's merged and check latest models to ensure that patterns work for GPU too.

File: src/common/transformations/src/transformations/common_optimizations/fuse_rotary_positional_embeddings.cpp (hunk 1 above)
Comment (Lyamin-Roman) [reply src/common/transformations/src/transformations/common_optimizations/fuse_rotary_positional_embeddings.cpp:656]: Because for some reason this complication leads to the detection of only 2/64 cases and it is easier to simplify the check than to add several more conditions

File: src/plugins/intel_gpu/tests/functional/shared_tests_instances/subgraph_tests/rotary_pos_emb.cpp [hunk 2]
```diff
@@ -0,0 +1,626 @@
...
+
+#include <string>
+
+using namespace ov::gen_pattern;
+using namespace ov;
+
+namespace ov {
+namespace test {
+
+inline void CheckNumberOfNodesWithType(std::shared_ptr<const ov::Model> function,
```
[the remark this answers is not in the document]
Comment (Lyamin-Roman) [reply src/plugins/intel_gpu/tests/functional/shared_tests_instances/subgraph_tests/rotary_pos_emb.cpp]: Done

**Review by usstq — APPROVED** [review]
LGTM
